Lib/test/test_subprocess.py

```diff
@@ -376,7 +376,9 @@ def _assert_cwd(self, expected_cwd, python_arg, **kwargs):
         # matches *expected_cwd*.
         p = subprocess.Popen([python_arg, "-c",
                               "import os, sys; "
-                              "sys.stdout.write(os.getcwd()); "
+                              "buf = sys.stdout.buffer; "
+                              "buf.write(os.getcwd().encode()); "
+                              "buf.flush(); "
```

Comment:
You could also force standard I/O encoding to use UTF-8 with the argument `env={**os.environ, 'PYTHONIOENCODING': 'utf-8'}`.